components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/Extensions.kt

```diff
@@ -0,0 +1,17 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.toolbar
+
+import java.text.NumberFormat
+import java.util.Locale
+
+// This is used for truncating URLs to prevent extreme cases from
+// slowing down UI rendering e.g. in case of a bookmarklet or a data URI.
+// https://github.com/mozilla-mobile/android-components/issues/5249
+const val MAX_URI_LENGTH = 25000
+
+fun getTrimmedUrl(url: String): String {
```

Comment:
I am wondering if this should be moved to one of our more generic support components since this is being used more widely outside this browser-toolbar component. One idea is support-ktx https://searchfox.org/mozilla-mobile/source/android-components/components/support/ktx/src/main/java/mozilla/components/support/ktx/kotlin/String.kt

We also want to make this an actual extension of Strings 

```suggestion
fun String.toTrimmedUrl(): String {
```

Reply:
Also, please add a kDoc

Reply:
Agree!

Reply:
I was kinda looking for this. Thank you!